README.md
<!--
 * @Author: ice
 * @Date: 2020-12-14 17:45:16
 * @LastEditTime: 2020-12-24 18:10:30
 * @LastEditors: Please set LastEditors
 * @Description: In User Settings Edit
 * @FilePath: /cra-template-rack/README.md
-->
# cra-template-rack

This is a template for [Create React App](https://github.com/facebook/create-react-app).

[![node](https://img.shields.io/node/v/cra-template-rack.svg?style=flat-square)](https://nodejs.org/en/)
![npm](https://img.shields.io/npm/dt/cra-template-rack?style=flat-square)
![Libraries.io dependency status for GitHub repo](https://img.shields.io/librariesio/github/poeticloud/cra-template-rack?style=flat-square)
![NPM](https://img.shields.io/npm/l/cra-template-rack?style=flat-square)

## usage

```sh
$ npx create-react-app my-app --template rack
```
## Views
![image](https://user-images.githubusercontent.com/8178166/114004851-9f95b300-9891-11eb-85f6-1fd176dae6e1.png)
![image](https://user-images.githubusercontent.com/8178166/114005069-ca800700-9891-11eb-8496-41805e6760fb.png)
![image](https://user-images.githubusercontent.com/8178166/114005109-d370d880-9891-11eb-9e7a-bf16496fd081.png)
## Highlights
> Compared with create-react-app's official template cra-template
- sass
- axios
- router
- redux
- prettier
- lint
- mock
- test
- dockerFile
- stipulate directory
## Links
> If you need typescript, you can go [Cra-template-rack-typescript](https://www.npmjs.com/package/cra-template-rack-typescript).

For more information, please refer to:

- [Getting Started](https://create-react-app.dev/docs/getting-started) – How to create a new app.
- [User Guide](https://create-react-app.dev) – How to develop apps bootstrapped with Create React App.


CHANGELOG.md
## [0.5.2](https://github.com/poeticloud/cra-template-rack/compare/v0.5.0...v0.5.2) (2021-04-12)


### Bug Fixes

* 非认证接口不走认证机制 ([1e714c2](https://github.com/poeticloud/cra-template-rack/commit/1e714c2904fbacdc9ddd3fa72f306a374953885f))



# [0.5.0](https://github.com/poeticloud/cra-template-rack/compare/v0.4.0...v0.5.0) (2021-03-08)


### Bug Fixes

* 无痛刷新token ([9389405](https://github.com/poeticloud/cra-template-rack/commit/9389405b64886903cd938f9003a5521e09a3fc25))
* **account:** 当get请求过期刷新token后重载页面 ([590f758](https://github.com/poeticloud/cra-template-rack/commit/590f7582933be2dd013c04460998abbea6bb9358))


### Features

* **all:** changelog ([3f82405](https://github.com/poeticloud/cra-template-rack/commit/3f8240556f4bd62db5cda373f5104767f509a11d))



# [0.4.0](https://github.com/poeticloud/cra-template-rack/compare/v0.3.0...v0.4.0) (2021-02-20)


### Features

* **all:** commit符合angular规范 ([e6ed5cd](https://github.com/poeticloud/cra-template-rack/commit/e6ed5cd615a230ffb38d95750341a7289bb8584e))


* <refactor>(<视图层>): <重构项目基础架构> ([f0e0645](https://github.com/poeticloud/cra-template-rack/commit/f0e064550d7fbb76ef4f4ecfccce7eebf5ca1339))


### BREAKING CHANGES

* 由于发布间隔比较久，与上个版本不兼容，建议运用新版本



# [0.3.0](https://github.com/poeticloud/cra-template-rack/compare/0.2.2...v0.3.0) (2020-12-22)



## [0.2.2](https://github.com/poeticloud/cra-template-rack/compare/v0.2.1...0.2.2) (2020-12-21)



## [0.2.1](https://github.com/poeticloud/cra-template-rack/compare/v0.1.2...v0.2.1) (2020-12-16)



## 0.1.2 (2020-12-15)





template/src/mock/db.js
/*
 * @Author: your name
 * @Date: 2020-12-15 17:36:19
 * @LastEditTime: 2020-12-16 15:58:22
 * @LastEditors: Please set LastEditors
 * @Description: In User Settings Edit
 * @FilePath: /my-app/mock/db.js
 */
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

// This sets the mock adapter on the default instance
var mock = new MockAdapter(axios);

// Mock GET request to /users when param `searchText` is 'John'
// arguments for reply are (status, data, headers)
mock
  .onGet('/svc')
  .reply(200, {
    count: 561,
    results: [
      { id: 'aPE6DlkD', title: 'wlytest1', version: '0.0.1' },
      {
        id: 'b9RpYEPa',
        title: 'Apis Editor Backend API Document',
        version: '0.1.0',
      },
      { id: 'OZE8QKMA', title: '用友累死了', version: '0.0.1' },
      { id: 'ZzEGBR8J', title: '自定义服务3', version: '0.0.1' },
      { id: 'ZdEb3Kj8', title: '用友累死了', version: '0.0.1' },
      { id: 'a0RL3l8p', title: '自定义服务4', version: '0.0.1' },
      { id: 'YqlaNRV3', title: '自定义服务2', version: '0.0.1' },
      { id: 'ZxRyGKAD', title: '自定义服务5', version: '0.0.1' },
      { id: 'VWlkpR1a', title: '自定义服务6', version: '0.0.1' },
      { id: '3bEmOlZk', title: 'Swagger Petstore', version: '1.0.0' },
      { id: 'D2l4GK3y', title: '鹏飞服务', version: '0.0.2' },
    ],
  })
  .onPost('/svc')
  .reply(200)
  .onPut(/\/svc\/[^/\\\s]+/)
  .reply(200)
  .onDelete(/\/svc\/[^/\\\s]+/)
  .reply(200)
  .onPost('/auth/obtain-token')
  .reply(200, {
    access_token:
      'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJhZG1pbkBwb2V0aWNsb3VkLmNvbSIsImV4cCI6MTYxMjQzNjczMCwidHlwIjoiYWNjZXNzIn0.GV8pCIRk6sOB0VZzhZjqzT_w1x5zHzah1RHzdA75WR8',
    refresh_token:
      'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJhZG1pbkBwb2V0aWNsb3VkLmNvbSIsImV4cCI6MTYxNzUxMDE0MCwidHlwIjoicmVmcmVzaCJ9.hic0ezaq6lwhk5f-LzEgI9PpKBUgMMBDhWvO8y3lDKI',
  })
  .onPost('/token/refresh')
  .reply(200, {
    access_token:
      'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJyZnVnbGo3ODk2MEBjaGFjdW8ubmV0IiwiZXhwIjoxNjEyNDMyMjQxLCJ0eXAiOiJhY2Nlc3MifQ.kRbL403yS_VOrizRTtQF5QrsXGbRGoxcYvlmNEBjsuU',
  })
  .onGet('/me')
  .reply(200, {
    created_at: '2021-02-01T11:03:26.408511+00:00',
    updated_at: '2021-02-02T03:57:13.713115+00:00',
    id: 47,
    email: '1214475704@qq.com',
    name: '唐助手',
    role: '学生',
    is_active: true,
    can_manage_ana_tools: true,
    lab: {
      created_at: '2021-02-01T10:51:13.041350+00:00',
      updated_at: '2021-02-01T10:52:03.360143+00:00',
      id: 59,
      name: '唐氏洗脚城',
      is_active: true,
      shared_projects: [],
    },
    created_projects: [],
  })
  .onPut('/me')
  .reply(200, {
    message: 'OK',
  })
  .onPost('/resend-email/find-password')
  .reply(200, {
    message: 'OK',
  })
  .onPost('/reset-password')
  .reply(200, {
    message: 'OK',
  })
  .onPost('/resend-email/register')
  .reply(200, {
    message: 'OK',
  });
